deploy/role.yaml

```diff
@@ -1,7 +1,7 @@
 apiVersion: rbac.authorization.k8s.io/v1
 kind: Role
 metadata:
-  name: vitess-operator-subcontroller
+  name: vitess-operator-backup-storage
```

Comment:
This is from a previous bad commit, we had inconsistencies, sometimes it was `-subcontroller` and sometimes `-backup-storage`.

Modifying everything back to `-backup-storage`.